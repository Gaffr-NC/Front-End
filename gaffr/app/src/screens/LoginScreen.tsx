import React, { Component } from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Alert,
  AsyncStorage,
  TouchableOpacity,
  Image,
  Keyboard,
  TouchableWithoutFeedback
} from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import { UserCredential } from '@firebase/auth-types';
import { getUserById } from '../utils';
import firebase, { FirebaseError } from 'firebase';
import { DocumentData } from '@firebase/firestore-types';

interface States {
  email: String;
  password: String;
}

interface Props {
  navigation: NavigationScreenProp<any, any>;
}

export default class Login extends Component<Props, States> {
  public state = {
    email: '',
    password: ''
  };

  static navigationOptions = {
    title: 'Welcome'
  };

  DismissKeyboard = ({ children }: any) => (
    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
      {children}
    </TouchableWithoutFeedback>
  );

  render() {
    return (
      <this.DismissKeyboard>
        <View style={styles.loginContainer}>
          <View style={styles.headerContainer}>
            <Image
              style={styles.logo}
              source={{ uri: 'https://i.imgur.com/NH0xhhe.png' }}
            />
            <Text style={styles.headerText}>
              It's time to move forward, and it's time to move in...
            </Text>
          </View>

          <View style={styles.formContainer}>
            <TextInput
              style={styles.loginForm}
              placeholder="Email Address"
              value={this.state.email}
              onChangeText={(text: String) => this.setState({ email: text })}
              autoCapitalize="none"
            />
            <TextInput
              style={styles.loginForm}
              placeholder="Password"
              value={this.state.password}
              onChangeText={(text: String) => this.setState({ password: text })}
              autoCapitalize="none"
              secureTextEntry
            />
            <TouchableOpacity
              style={styles.loginButton}
              onPress={this.handleLogInPress}
            >
              <Text style={{ alignSelf: 'center', color: '#ffffff' }}>
                LOG IN
              </Text>
            </TouchableOpacity>
          </View>

          <View style={styles.signupContainer}>
            <Text>New user?</Text>
            <TouchableOpacity
              style={styles.signupButton}
              onPress={() => this.props.navigation.navigate('userType')}
            >
              <Text style={{ alignSelf: 'center', color: '#ffffff' }}>
                SIGN UP
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </this.DismissKeyboard>
    );
  }

  handleLogInPress = (): void => {
    const { email, password } = this.state;
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(async (user: UserCredential) => {
        if (user.user) {
          const { uid } = user.user;
          const { navigate } = this.props.navigation;
          const tenant: DocumentData | undefined = await getUserById(
            uid,
            'tenants'
          );
          if (tenant) {
            await AsyncStorage.setItem('uid', uid);
            await AsyncStorage.setItem('userType', 'tenants');
            navigate('TenantApp', { uid });
          } else {
            const landlord: DocumentData | undefined = await getUserById(
              uid,
              'landlords'
            );
            if (landlord) {
              await AsyncStorage.setItem('uid', uid);
              await AsyncStorage.setItem('userType', 'landlords');
              navigate('Properties', { uid });
            }
          }
        }
      })
      .catch((err: FirebaseError) => {
        Alert.alert('Invalid email/password');
      });
  };
}

const styles = StyleSheet.create({
  loginContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
    margin: 0,
    color: '#0B4F6C',
    backgroundColor: '#dcd1e8'
  },
  headerContainer: {
    alignItems: 'center',
    width: '100%',
    margin: 0
  },
  logo: {
    alignItems: 'center',
    height: 150,
    width: 200
  },
  headerText: {
    textAlign: 'center',
    marginBottom: 20,
    alignItems: 'center',
    fontSize: 12,
    padding: 0
  },
  formContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    padding: 0,
    margin: 0
  },
  loginForm: {
    margin: 5,
    width: '90%',
    padding: 5,
    backgroundColor: '#f9f4f5',
    borderRadius: 10
  },
  loginButton: {
    backgroundColor: '#502F4C',
    color: '#ffffff',
    margin: 5,
    width: '90%',
    padding: 5,
    borderRadius: 10
  },
  signupContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    padding: 70,
    margin: 0
  },
  signupButton: {
    backgroundColor: '#502F4C',
    margin: 0,
    marginTop: 10,
    width: '75%',
    padding: 20,
    borderRadius: 10
  }
});
