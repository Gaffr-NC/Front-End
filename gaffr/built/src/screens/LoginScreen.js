"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const react_native_1 = require("react-native");
const firebase_1 = __importDefault(require("firebase"));
class Login extends react_1.Component {
    constructor() {
        super(...arguments);
        this.state = {
            email: '',
            password: ''
        };
        this.logIn = () => {
            console.log(this.props);
            this.props.navigation.navigate('TenantApp');
        };
        this.handleLogInPress = () => {
            console.log('PRESSED LOG IN');
            const { email, password } = this.state;
            console.log(email, password);
            firebase_1.default
                .auth()
                .signInWithEmailAndPassword(email, password)
                .then((user) => {
                this.props.navigation.navigate('TenantApp');
                console.log(user, 'UUUUSER');
            })
                .catch((err) => {
                console.log(err);
                react_native_1.Alert.alert('Invalid email/password');
            });
        };
    }
    render() {
        return (react_1.default.createElement(react_native_1.View, { style: { flex: 1, alignItems: 'center', justifyContent: 'center' } },
            react_1.default.createElement(react_native_1.Text, null, "Log in or sign up!"),
            react_1.default.createElement(react_native_1.TextInput, { style: styles.inputs, placeholder: "email...", value: this.state.email, onChangeText: (text) => this.setState({ email: text }), autoCapitalize: "none" }),
            react_1.default.createElement(react_native_1.TextInput, { style: styles.inputs, placeholder: "password...", value: this.state.password, onChangeText: (text) => this.setState({ password: text }), autoCapitalize: "none", secureTextEntry: true }),
            react_1.default.createElement(react_native_1.Button, { title: "LOG IN", onPress: this.handleLogInPress }),
            react_1.default.createElement(react_native_1.Button, { title: "SIGN UP", onPress: () => this.props.navigation.navigate('userType') })));
    }
}
Login.navigationOptions = {
    title: 'Log-In'
};
exports.default = Login;
const styles = react_native_1.StyleSheet.create({
    inputs: {
        backgroundColor: 'powderblue',
        margin: 10,
        width: '90%',
        padding: 10
    }
});
