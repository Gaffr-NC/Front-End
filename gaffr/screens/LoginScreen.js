import React, { Component } from "react";
import { View, Text, Button, TextInput, StyleSheet, Alert } from "react-native";
import firebase from 'firebase';

export default class Login extends Component {
  state = {
    email: "",
    password: ""
  };

  static navigationOptions = {
    title: "Log-In"
  };

  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>Log in or sign up!</Text>
        <TextInput
          style={styles.inputs}
          label="email"
          placeholder="email..."
          value={this.state.userInput}
          onChangeText={text => this.setState({ email: text })}
          autoCapitalize="none"
        />
        <TextInput
          style={styles.inputs}
          label="password"
          placeholder="password..."
          value={this.state.userInput}
          onChangeText={text => this.setState({ password: text })}
          autoCapitalize="none"
          secureTextEntry
        />
        <Button
          title="LOG IN"
          onPress={this.handleLogInPress}
        />
        <Button
          title="SIGN UP"
          onPress={() => this.props.navigation.navigate("signUp")}
        />
      </View>
    );
  }

  logIn = () => {
    console.log(this.props)
    this.props.navigation.navigate("TenantApp")
  }

  handleLogInPress = () => {
    console.log("PRESSED LOG IN")
    const { email, password } = this.state;
    console.log(email, password);
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then((user) => {
        this.props.navigation.navigate("TenantApp")
        console.log(user, "UUUUSER");
      })
      .catch(err => {
        console.log(err)
        Alert.alert("Invalid email/password")
      });
  };
}



const styles = StyleSheet.create({
  inputs: {
    backgroundColor: "powderblue",
    margin: 10,
    width: "90%",
    padding: 10,
  }
});
