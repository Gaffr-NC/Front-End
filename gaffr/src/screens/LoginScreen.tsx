import React, { Component } from 'react';
import { View, Text, Button, TextInput, StyleSheet, Alert } from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import firebase from 'firebase';

interface States {
  email: String;
  password: String;
}

interface Props {
  navigation: NavigationScreenProp<any, any>;
}

export default class Login extends Component<Props, States> {
  public state = {
    email: '',
    password: ''
  };

  static navigationOptions = {
    title: 'Log-In'
  };

  render() {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>Log in or sign up!</Text>
        <TextInput
          style={styles.inputs}
          placeholder="email..."
          value={this.state.email}
          onChangeText={text => this.setState({ email: text })}
          autoCapitalize="none"
        />
        <TextInput
          style={styles.inputs}
          placeholder="password..."
          value={this.state.password}
          onChangeText={text => this.setState({ password: text })}
          autoCapitalize="none"
          secureTextEntry
        />
        <Button title="LOG IN" onPress={this.handleLogInPress} />
        <Button
          title="SIGN UP"
          onPress={() => this.props.navigation.navigate('userType')}
        />
      </View>
    );
  }

  logIn = () => {
    console.log(this.props);
    this.props.navigation.navigate('TenantApp');
  };

  handleLogInPress = () => {
    console.log('PRESSED LOG IN');
    const { email, password } = this.state;
    console.log(email, password);
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then((user: any) => {
        this.props.navigation.navigate('TenantApp');
        console.log(user, 'UUUUSER');
      })
      .catch((err: any) => {
        console.log(err);
        Alert.alert('Invalid email/password');
      });
  };
}

const styles = StyleSheet.create({
  inputs: {
    backgroundColor: 'powderblue',
    margin: 10,
    width: '90%',
    padding: 10
  }
});
