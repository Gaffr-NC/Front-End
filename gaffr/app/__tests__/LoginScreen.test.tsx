import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../src/screens/LoginScreen';

let props: any;
it('LoginScreen renders without crashing', () => {
  const rendered = renderer.create(<LoginScreen {...props} />).toJSON();
  expect(rendered).toBeTruthy();
});
it('Tests LoginScreen against snapshot', () => {
  const tree = renderer.create(<LoginScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
