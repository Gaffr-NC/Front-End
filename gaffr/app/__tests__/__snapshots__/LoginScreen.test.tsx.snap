// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests LoginScreen against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dcd1e8",
      "color": "#0B4F6C",
      "flex": 1,
      "justifyContent": "center",
      "margin": 0,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 0,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.imgur.com/NH0xhhe.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 12,
          "padding": 0,
        }
      }
    >
      It's time to move forward, and it's time to move in...
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "padding": 0,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email Address"
      style={
        Object {
          "backgroundColor": "#f9f4f5",
          "borderRadius": 10,
          "margin": 5,
          "padding": 5,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#f9f4f5",
          "borderRadius": 10,
          "margin": 5,
          "padding": 5,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#502F4C",
          "borderRadius": 10,
          "color": "#ffffff",
          "margin": 5,
          "opacity": 1,
          "padding": 5,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
          }
        }
      >
        LOG IN
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "padding": 70,
        "width": "100%",
      }
    }
  >
    <Text>
      New user?
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#502F4C",
          "borderRadius": 10,
          "margin": 0,
          "opacity": 1,
          "padding": 20,
          "width": "55%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;
